junit-runtime/src/main/scala/com/novocode/junit/RichLogger.scala

```diff
@@ -0,0 +1,171 @@
+package com.novocode.junit
+
+import sbt.testing._
+
+import scala.collection.mutable
+
+import Ansi._
+
+final class RichLogger private (loggers: Array[Logger], settings: RunSettings) {
+
+  private[this] val currentTestClassName = new mutable.Stack[String]()
+
+  def this(loggers: Array[Logger], settings: RunSettings,
+      testClassName: String) = {
+    this(loggers, settings)
+    currentTestClassName.push(testClassName)
+  }
+
+  def pushCurrentTestClassName(s: String): Unit =
+    currentTestClassName.push(s)
+
+  def popCurrentTestClassName(): Unit = {
+    if (currentTestClassName.size > 1)
+      currentTestClassName.pop()
+  }
+
+  def debug(s: String): Unit = {
+    for (l <- loggers) {
+      if (settings.color && l.ansiCodesSupported())
+        l.debug(s)
+      else
+        l.debug(filterAnsi(s))
+    }
+  }
+
+  def error(s: String): Unit = {
+    for (l <- loggers) {
+      if (settings.color && l.ansiCodesSupported())
+        l.error(s)
+      else
+        l.error(filterAnsi(s))
+    }
+  }
+
+  def error(s: String, t: Throwable): Unit = {
+    error(s)
+    if (t != null && (settings.logAssert || !t.isInstanceOf[AssertionError]))
+      logStackTrace(t)
+  }
+
+  def info(s: String): Unit = {
+    for (l <- loggers) {
+      if (settings.color && l.ansiCodesSupported())
+        l.info(s)
+      else
+        l.info(filterAnsi(s))
+    }
+  }
+
+  def warn(s: String): Unit = {
+    for (l <- loggers) {
+      if (settings.color && l.ansiCodesSupported()) l.warn(s)
+      else l.warn(filterAnsi(s))
```

Comment:
if/else in statement position ->
```scala
if (settings.color && l.ansiCodesSupported())
  l.warn(s)
else
  l.warn(filterAnsi(s))
```
However, I suggest factoring out that test as
```scala
def filterAnsiIfNeeded(s: String): String =
  if (settings.color && l.ansiCodesSupported()) s
  else filterAnsi(s)
```
and then do
```scala
l.warn(filterAnsiIfNeeded(s))
```
and use the same thing in `info`, `error` and `debug`